docs/howtos/optax_update_guide.rst

```diff
@@ -0,0 +1,263 @@
+Optax Update Guide
+==================
+
```

Comment:
I think a short intro is in order here, e.g. "The built-in `flax.optim` package is being deprecated, and we instead recommend all users use the [Optax](link to github)] library. Follow this guide to upgrade your codebase."